src/metabase/query_processor/util.clj

```diff
@@ -30,7 +30,7 @@
    can access the default value."
   [{{:keys [executed-by query-hash], :as _info} :info, query-type :type}]
   (str "Metabase" (when executed-by
-                    (assert (instance? (Class/forName "[B") query-hash))
+                    (assert (bytes? query-hash) "If info includes executed-by it should also include query-hash")
```

Comment:
Unrelated but I improved the error message here if you're missing `query-hash` which I guess is required if you also have `executed-by`.